modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/idp/mgt/IdentityProviderMgtServiceTestCase.java

```diff
@@ -543,4 +743,37 @@ public void testDeleteIdP() throws Exception {
         Assert.assertNull(idp, "Deleting idp failed");
     }
 
+    private void addIdpForPagination() throws Exception {
+
+        String testIdpDescription = "This is second identity provider ";
+        String testIdpRealmId = "localhost";
+        String testFedAuthDispName = "openid";
+        IdentityProvider idProvider = new IdentityProvider();
+        // Set idp information.
+        idProvider.setHomeRealmId(testIdpRealmId);
+        idProvider.setEnable(true);
+        idProvider.setIdentityProviderDescription(testIdpDescription);
+        idProvider.setIdentityProviderName(testIdpNameSearch);
+        idProvider.setCertificate(sampleCertificate);
+        idProvider.setFederationHub(false);
+        idProvider.setPrimary(false);
+        // Add federated authentication configuration.
+        FederatedAuthenticatorConfig authConfig = new FederatedAuthenticatorConfig();
+        authConfig.setDisplayName(testFedAuthDispName);
+        authConfig.setEnabled(true);
+        authConfig.setName(testFedAuthName);
+        // Add new identity provider.
+        idpMgtServiceClient.addIdP(idProvider);
+        // Check adding idp success.
+        IdentityProvider addedIdp = idpMgtServiceClient.getIdPByName(testIdpName);
+        Assert.assertNotNull(addedIdp, "addIdP or getIdPByName failed");
```

Comment:
don't add asserts to private methods. Move them to where this private utility method is being used.